h5p-server/express.js

```diff
@@ -0,0 +1,279 @@
+const bodyParser = require("body-parser");
+const upload = require("express-fileupload");
+const cors = require("cors");
+const H5PServer = require("@lumieducation/h5p-server");
+const { h5pAjaxExpressRouter } = require("@lumieducation/h5p-express");
+const dotenv = require("dotenv");
+const cron = require("node-cron");
+const morgan = require("morgan");
+
+const { editorRenderer } = require("./renderers/editor");
+const { playerRenderer } = require("./renderers/player");
+const { adminRenderer } = require("./renderers/admin");
+const hooks = require("./custom/hooks");
+import MateriaTempFileStorage from "./interfaces/MateriaTempFileStorage.js";
+
+const app = express();
+
+const port = 3000;
+
+// if in development use '.env.local' file, else use '.env' file
+const envFile = process.env.ENVIRONMENT == "prod" ? `.env` : ".env.local";
+dotenv.config({ path: path.resolve(envFile) });
+
+function User() {
+  this.id = "1";
+  this.name = "Firstname Surname";
+  this.canInstallRecommended = true;
+  this.canUpdateAndInstallLibraries = true;
+  this.canCreateRestricted = true;
+  this.type = "local";
+  this.email = "test@example.com";
+}
+
+let h5pEditor = {};
+let h5pPlayer = {};
+
+const setupConfig = (resolve, reject) => {
+  new H5PServer.H5PConfig(
+    new H5PServer.fsImplementations.JsonStorage(path.resolve("h5p-config.json"))
+  )
+    .load()
+    .then(config => {
+      resolve(config);
+    });
+};
+
+const setupPlayerAndEditor = config => {
+  // using cached library storage to cache most common calls to the libraries, improving performance
+  // also using custom tempfilestorage to automate uploading media to materia from h5p server
+  const editor = new H5PServer.H5PEditor(
+    new H5PServer.fsImplementations.InMemoryStorage(),
+    config,
+    // new H5PServer.fsImplementations.FileLibraryStorage(
+    //   path.resolve("h5p/libraries")
+    // ),
+    new H5PServer.cacheImplementations.CachedLibraryStorage(
+      new H5PServer.fsImplementations.FileLibraryStorage(
+        path.resolve("h5p/libraries")
+      )
+    ),
+    new H5PServer.fsImplementations.FileContentStorage(
+      path.resolve("h5p/content")
+    ),
+    // new H5PServer.fsImplementations.DirectoryTemporaryFileStorage(
+    //   path.resolve("h5p/temporary-storage")
+    // )
+    new MateriaTempFileStorage(),
+    undefined,
+    undefined,
+    {
+      customization: {
+        global: {
+          scripts: ["/custom/fullscreen.js"]
+        },
+        alterLibrarySemantics: hooks.alterLibrarySemanticsHook
+      }
+    }
+  );
+
+  const player = new H5PServer.H5PPlayer(
+    editor.libraryStorage,
+    editor.contentStorage,
+    config
+  );
+
+  return [editor, player];
+};
+
+const setupServer = ([editor, player]) => {
+  h5pEditor = editor;
+  h5pPlayer = player;
+
+  const setupServerPromise = (resolve, reject) => {
+    // TODO replace whitelist with hosts from context env var
+    app.use(
+      cors({
+        origin: [
```

Comment:
Same as previous comment - these URLs will have to come from a config somewhere.